### PR TITLE
Fix bad agg output for custom types

### LINKED ISSUE #1798: Fix `array_agg` on money columns
`array_agg` on a money column looks pretty weird

image

### PR DESCRIPTION
**Pull request by dmos62**
Fixes #1798

Previous PR addressing this issue had a bug. This fixes it. It also adds a test (which the mentioned PR didn't do).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/tests/conftest.py [hunk 1]
```diff
@@ -251,6 +251,7 @@ def uris_table_obj(engine_with_uris, uris_table_name):
             engine,
             conn,
             uri_type,
+            metadata=metadata,
```
Comment (dmos62) [inline db/tests/conftest.py:254]: Unrelated to PR; generally good idea.

**Comment by mathemancer** [comment]
@dmos62 The tests are timing out on this branch; I'm not sure why.

**Comment by dmos62** [comment]
Marked the test for skipping. Not happy about this, but I don't feel like going down this rabbit hole right now.

**Comment by dmos62** [comment]
Brent is away right now on a 30 hour long flight, and this has been pending review for a long time. Also, Pavish's review here is probably superfluous. I'd like to not postpone merging this any longer.

I'd say that the review of this PR is a formality, because the PR is very simple. It fixes a basic bug, where we [forgot to assign the result of a function to a variable](https://github.com/centerofci/mathesar/pull/2074/files#diff-d4845f55bff3624db29baf018bf5105ce0f0beb6222ff4cf61c3f88f152a4797L389). It also provides a test for the functionality it fixes, so I'm confident that it's correct.

Because of above reasons, I'm merging without review being completed.

**Pull request merged**